archinstall/lib/installer.py

```diff
@@ -442,6 +441,10 @@ def add_bootloader(self, bootloader='systemd-bootctl'):
 				# Fallback, try creating the boot loader without touching the EFI variables
 				SysCommand(f'/usr/bin/arch-chroot {self.target} bootctl --no-variables --path=/boot install')
 
+			# Ensure that the /boot/loader directory exists before we try to create files in it
+			if not os.path.exists(f'{self.target}/boot/loader'):
+				os.makedirs(f'{self.target}/boot/loader')
+
```

Comment:
`bootctl` creates these directories. If it doesn't we should report it upstream and keep this is a patch with a `# TODO: Remove once <upstream> is fixed`.

Reply:
See #541 - in the case of these two installs by Steller in our Discord these dirs did NOT seem to be present.

![image](https://user-images.githubusercontent.com/277927/120636497-b4c73200-c43b-11eb-9d00-6b1393772ffe.png)
![image](https://user-images.githubusercontent.com/277927/120636513-b7c22280-c43b-11eb-96b8-18a4985530e7.png)

Reply:
This was using an existing /mnt by the way.

Reply:
I'm worried that `/boot` might have been incorrectly mounted, and `/` mounting over the location in some way.
Or out-of-disk size wize on `/boot` to create folders (more unlikely). I would love to get more debugging in to really know why this happens rather than trying to fix it hot-patch style.

But that said, introducing this *shouldn't* hurt, and we can remove it later when we have all the debugging in place to catch these things.